server/Episememe.Application.Tests/FeatureTests/FileUpload/WhenUploadingFile.cs

```diff
@@ -0,0 +1,193 @@
+﻿using Episememe.Application.Features.FileUpload;
+using Episememe.Application.Interfaces;
+using Episememe.Domain.Entities;
+using Episememe.Infrastructure.Database;
+using Episememe.Infrastructure.FileSystem;
+using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Internal;
+using Microsoft.Data.Sqlite;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.IO;
+using System.IO.Abstractions.TestingHelpers;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using Xunit;
+
+namespace Episememe.Application.Tests.FeatureTests.FileUpload
+{
+    public class WhenUploadingFile : IDisposable
+    {
+        private readonly string _givenMediaInstanceId = "abcdefgh";
+        private readonly DateTime _givenDate = new DateTime(2010, 1, 1);
+
+        private readonly MockFileSystem _fileSystemMock;
+        private readonly IFileStorage _fileStorageMock;
+        private readonly IWritableApplicationContext _contextMock;
+        private readonly Mock<ITimeProvider> _timeProviderMock;
+        private readonly Mock<IMediaIdProvider> _mediaIdProviderMock;
+
+        private readonly DbConnection _connection;
+
+        public WhenUploadingFile()
+        {
+            _fileSystemMock = new MockFileSystem();
+            var fileStorageSettings = Options.Create(new FileStorageSettings()
+            {
+                RootDirectory = ""
+            });
+            _fileStorageMock = new FileStorage(fileStorageSettings, _fileSystemMock);
+            (_contextMock, _connection) = GetInMemoryDatabaseContext();
+
+            _timeProviderMock = new Mock<ITimeProvider>();
+            _timeProviderMock.Setup(provider => provider.GetUtc())
+                .Returns(_givenDate);
+
+            _mediaIdProviderMock = new Mock<IMediaIdProvider>();
+            _mediaIdProviderMock.Setup(provider => provider.Generate())
+                .Returns(_givenMediaInstanceId);
+        }
+
+        [Fact]
+        public void GivenNewMedia_MediaIsUploadedSuccessfully()
```

Comment:
1. This is hardly a unit test. There are 4 assertions here, and 2 separate things are actually tested - 1) whether an expected file is created and 2) an expected entry is added to a database. You can split that into at least 2 tests.
2. Naming convention: **Given** part says little about what's happening in this test. Something like *GivenNewMediaIsUploadedWithoutErrors* is much more descriptive. And then, *ThenDatabaseContainsExpectedMediaInstance* or *ThenFileSystemContainsExpectedFile*